### PR TITLE
Print version and exit if --version is present

### PR DESCRIPTION
**Pull request by metanet**
Store the version in pervane/version.txt. It is used by both the app
itself and the build tool.

### CODE REVIEW

File: pervane/serve.py [hunk 1]
```diff
@@ -55,6 +55,8 @@
 from werkzeug.security import generate_password_hash, check_password_hash
 from werkzeug.utils import secure_filename
 
+from pathlib import Path
```
Comment (hakanu) [inline pervane/serve.py:58]: let's not directly use class, instead let's use it from the module.
import pathlib
...
pathlib.Path

File: pervane/serve.py [hunk 2]
```diff
@@ -102,10 +104,22 @@ def str2bool(v):
...
     '--debug', type=str2bool, dest='debug', default=False,
     help='Show debug logs')
+parser.add_argument(
+    '--version', action='store_true', dest='version', default=False,
+    help='Show version and exit')
 
 args = parser.parse_args()
+if args.version:
+  # early return if only the version is asked.
+  version_file = str(Path(__file__).resolve().parent / "version.txt")
```
Comment (hakanu) [inline pervane/serve.py:114]: use os.path.sep instead of hardcoding separator.

Comment (hakanu) [inline pervane/serve.py:114]: even better use os.path.join to create paths

File: pervane/serve.py [hunk 3]
```diff
@@ -102,10 +104,22 @@ def str2bool(v):
...
+  with open(version_file, "r") as version_fh:
+    version = version_fh.read()
+    print('Pervane ', version)
+    sys.exit()
+
 # Log may not show up in non-debug mode.
```
Comment (hakanu) [inline pervane/serve.py:120]: Let's take this before version printing for better debuggability.

Comment (hakanu) [inline pervane/serve.py:120]: Let's take this before version printing for better debuggability.

Comment (metanet) [reply pervane/serve.py:120]: then, we get the following output for `pervane --version` which I am not sure of
```
$ python3 pervane/serve.py --version
loaded args %s Namespace(allow_multi_user=False, cache_seconds=2, debug=False, front_page_message='Welcome to Pervane! This is default welcome message.', host='0.0.0.0', ignore_patterns=['env/.*', '.git', '.*.swp', '.*.pyc', '__pycache__', '.allmark', '_vnote.json', '!!!meta.json', 'PaxHeader', '.nojekyll', '.*.sqlite'], note_extensions=['.txt', '.md', '.markdown'], password=None, port='5000', root_dir='/home/basri/dev/personal/repos/notes', username=None, version=True)
Pervane  0.0.52
```

Comment (hakanu) [reply pervane/serve.py:119]: nvm

**Pull request merged**